platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/PeerInfo.java

```diff
@@ -19,18 +19,22 @@
 import com.swirlds.common.platform.NodeId;
 import com.swirlds.platform.roster.RosterUtils;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import java.security.cert.Certificate;
 
 /**
- * A record representing a peer's network information.
+ * A record representing a peer's network information.  If the certificate is not null, it must be encodable as a valid
+ * X509Certificate.  It is assumed the calling code has already validated the certificate.
  *
  * @param nodeId             the ID of the peer
  * @param hostname           the hostname (or IP address) of the peer
- * @param signingCertificate the certificate used to validate the peer's TLS certificate
+ * @param signingCertificate the certificate used to validate the peer's TLS certificate, or null.
  */
-public record PeerInfo(@NonNull NodeId nodeId, @NonNull String hostname, @NonNull Certificate signingCertificate) {
+public record PeerInfo(@NonNull NodeId nodeId, @NonNull String hostname, @Nullable Certificate signingCertificate) {
```

Comment:
why make it nullable?

Reply:
You're right, after examining the code, the entries that are null are dropped from the list and the record doesn't need to support null.

Reply:
When I make the field `NonNull`, the IDE complains of possible null because it can't reason about the filter applied earlier in the stream.    Making it nullable also forces the code closest to use to deal with the potential of no certificate or bad certificate.

Reply:
I disagree with this decision. Checks should be done once, so that we don't have to re-check lots of times downstream. A list of peers should be ones with valid certificates, so we should ensure that when constructing the list and then trust if from then on. With this approach, we just add a bunch of boilerplate code with no real purpose.

Reply:
I have restored it to being `@NonNull`